portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -74,28 +83,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       commentEntity.setProperty("name", name);
       commentEntity.setProperty("comment", comment);
       commentEntity.setProperty("timestamp", System.currentTimeMillis());
+      commentEntity.setProperty("length", comment.length());
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(commentEntity);
 
-      // pause the thread to ensure data is in DB before page refresh
-      // could not find equivalent of async call to datastore (tried transactions
-      // but couldn't get them to work)
-      try {
-        Thread.sleep(10);
-      } catch (InterruptedException e) {
-        e.printStackTrace();
-      }
-
       response.sendRedirect("/index.html#Comments");
   }
 
   /**
-   * Converts the hardcoded ArrayList to JSON using Gson Java library
+   * Converts the List of Comment objects to JSON using Gson Java library
    */
   private String convertToJson(List<Comment> arr) {
     Gson gson = new Gson();
     String json = gson.toJson(arr);
     return json;
   }
+
+/**
+ * Prepares a query with the user-inputed sorting order
+ */
+  private Query prepareQuery(String sortingOrder) {
+      // set to the default case
+      Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+      // Given user input, format the query
+      switch (sortingOrder) {
+          case "timestamp_descending":
+            break;
```

Comment:
i get an error that then it might return an unitialized value.... my thoughts were I could make each case return instead of breaking or to just remove the "timestamp_descending" case from the switch statement (have it act as a default declaration)

which of these approaches do you prefer?

Reply:
you probably need a default case. you can have both the timesamp_descending case and default case do the same thing by omitting the break (example about halfway down https://www.geeksforgeeks.org/switch-statement-in-java/)